### DISCUSSION TITLE
ADD MANAGER AND FREEZE ADDRESS AS ALGORAND-ONLY OPTIONS TO LAUNCHTOKEN

### DISCUSSION THREAD

**Discussion by temptemp3** [reach-sh/reach-lang#1260] [Ideas]
Hello,
I have use case that requires setting the manager and freeze address when launching a non-network token. Currently, clawback is provided as a Algorand-only option with the default of and empty string. Can we add manager and freeze address as well?
I don't want to be forced to use algosdk and reinvent the wheel building asset create transactions.
Thx!

**Comment by temptemp3** [comment]
Nevermind. I figured something out.

**Reply by jeapostrophe** [reply]
This will be fixed by https://github.com/reach-sh/reach-lang/pull/1280

**Reply by jchaselubitz** [reply]
I see this was added in #1280 (awesome!!) and wanted to check if it's possible to recreate the behavior described here: https://developer.algorand.org/solutions/securities-and-permissioned-tokens/

Specifically: I want all transfers of an ASA to require permission from a smart contract. The tutorial linked above proposes doing that by freezing the ASA, setting the clawback address to a contract address, and then having that contract conduct all transfers. It suggests that the clawback address should be able to transfer frozen tokens.

When we @(@asolpshinning) try this in reach (sending tokens from the clawback address), we get this message: `asset frozen in recipient`

Could you explain why that is and if there is a way around it?

**Reply by temptemp3** [reply]
@jchaselubitz The asset is frozen and may be frozen by default. Good luck trying to use the clawback Algorand feature in a reach app ; )
You could unfreeze the asset but that wouldn't help you.
You could construct an asset revocation transaction but that would require you operate the create with the clawback address. 
You could could come up with a solution not requiring clawback and frozen assets.

**Reply by jchaselubitz** [reply]
Do you know if what we are trying to do is possible with algoSDK? I am just trying to get a sense of where the problem lies, as it seems this can be done (somehow) on Algorand.

**Reply by jeapostrophe** [reply]
I don't know anything about specifically how frozen accounts work. However, Reach does not support generating txns for any accounts other than the contract account AND it will not allow you to deal with funds that do not come from a `.pay`.

**Reply by Thomas-Bibby** [reply]
What about the manager address? When will this be added in?

**Reply by jeapostrophe** [reply]
You can follow #1440

**Comment by KiroAlgorand** [comment]
Hi. I have been trying to freeze my token authority but I can't also the Clawback, anyone can help me?
